addons/web/controllers/main.py

```diff
@@ -1434,24 +1465,31 @@ def content_type(self):
 
     def filename(self, base):
         return base + '.csv'
-
-    def from_data(self, fields, rows):
+    
+    def from_data(self, fields, rows, is_local_timezone):
+        """
+        Export the data to a csv file. 
+        Return: bytes 
+        """
         fp = io.BytesIO()
         writer = pycompat.csv_writer(fp, quoting=1)
-
         writer.writerow(fields)
-
         for data in rows:
             row = []
             for d in data:
-                # Spreadsheet apps tend to detect formulas on leading =, + and -
                 if isinstance(d, pycompat.string_types) and d.startswith(('=', '-', '+')):
                     d = "'" + d
-
+                if type(d) is str and is_local_timezone:
+                    parse_data = self._check_date(d)
+                    if parse_data and request._context.get('tz'):
+                        tz = pytz.timezone(request._context.get('tz'))
+                        d = (pytz.utc.localize(parse(d)).astimezone(tz)).strftime('%Y-%m-%d %H:%M:%S')
+                    elif parse_data and not request._context.get('tz'): 
+                        raise UserError(_("The user does not have a timezone set"))
```

Comment:
Can we re-word this? It's going to be an issue with the user's own account and should be made clear it is because they've chose to export in their local timezone.

Maybe: `Unable to export data in local time, no timezone is set against your user record.`